### PR TITLE
Initial commit of the PowerDNS Authoritative API

### PR DESCRIPTION
**Pull request by dasJ**
Yes, I am aware of the fact that you cannot create records, this will follow in another PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+| Name                                  | Default/Required   | Description                                         |
+|---------------------------------------|:------------------:|-----------------------------------------------------|
+| `pdns_auth_api_connect`               | :heavy_check_mark: | Connect to this URL (e.g. `http://127.0.0.1:1234`)  |
+| `pdns_auth_api_server`                | `localhost`        | Server instance to connect to                       |
+| `pdns_auth_api_key`                   | :heavy_check_mark: | API Key to use (may be empty if you don't have one) |
+| `pdns_auth_api_zones`                 | :heavy_check_marl: | List of DNS zones (see below)                       |
+| `pdns_auth_api_remove_unknown_zones:` | `false`            | Delete zones that are now known to this role        |
+
+#### DNS Zones
```
Comment (SlothOfAnarchy) [inline README.md]: you could use a higher level heading (`###`)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+| Name                 | Default/Required   | Description                                                                                                                                         |
+|----------------------|:------------------:|-----------------------------------------------------------------------------------------------------------------------------------------------------|
+| `name`               | :heavy_check_mark: | Name of this zone                                                                                                                                   |
+| `kind`               | `Master`           | Type of this zone (`Master`, `Slave`, or `Native`)                                                                                                  |
```
Comment (SlothOfAnarchy) [inline README.md:27]: why not `type` ?

File: defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+---
+pdns_auth_api_server: localhost
```
Comment (SlothOfAnarchy) [inline defaults/main.yml:2]: why are only some of the defaults defined here?

File: README.md (hunk 2 above)
Comment (dasJ) [reply README.md:27]: Because pdns calls it `kind`, and I don't want to reinvent the wheel

File: defaults/main.yml (hunk 3 above)
Comment (dasJ) [reply defaults/main.yml:2]: Because the other defaults are per-zone instead of per-host

**Comment by dasJ** [comment]
@SlothOfAnarchy Pls rereview

**Review by Scriptkiddi — APPROVED** [review]

**Pull request merged**